validation/src/test/scala/hmda/validation/rules/lar/quality/Q032Spec.scala

```diff
@@ -0,0 +1,36 @@
+package hmda.validation.rules.lar.quality
+
+import hmda.model.fi.lar.LoanApplicationRegister
+import hmda.parser.fi.FIGenerators
+import hmda.validation.rules.EditCheck
+import hmda.validation.rules.lar.{ BadValueUtils, LarEditCheckSpec }
+import org.scalacheck.Gen
+
+class Q032Spec extends LarEditCheckSpec with FIGenerators {
+
+  property("All lars with action taken types not equal to 1 must pass") {
+    forAll(larGen) { lar =>
+      whenever(lar.actionTakenType != 1) {
+        lar.mustPass
+      }
+    }
+  }
+
+  property("Lars with action taken = 1 must have different application and action taken dates") {
+    forAll(larGen, dateGen) { (lar, x) =>
```

Comment:
It seems like you might not need to use an independent `dateGen`, you could add 1 to `lar.actionTakenDate` or `lar.applicationDate` which I think would be a bit cleaner. This also applies for the following edit as well.